devito/ir/clusters/analysis.py

```diff
@@ -41,11 +49,19 @@ def callback(self, clusters, prefix):
         # Update `self.state`
         if retval:
             for c in clusters:
-                properties = self.state.properties.setdefault(c, {})
-                properties.setdefault(d, set()).update(retval)
+                c_properties = properties.setdefault(c, {})
+                c_properties.setdefault(d, set()).update(retval)
```

Comment:
Haven't look at this in a while so sorry for the comment, trying to remember. What's the point of line 46 to 54? It doesn't see to do anything.